clients/admin-ui/src/features/dataset/DatasetYamlForm.tsx

```diff
@@ -0,0 +1,117 @@
+import { Box, Button, Text, useToast } from "@fidesui/react";
+import { Form, Formik, FormikHelpers } from "formik";
+import yaml from "js-yaml";
+import { useRouter } from "next/router";
+
+import { CustomTextArea } from "../common/form/inputs";
+import { getErrorFromResult, isYamlException } from "../common/helpers";
+import { successToastParams } from "../common/toast";
+import { setActiveDataset, useCreateDatasetMutation } from "./dataset.slice";
+import { Dataset } from "./types";
+
+const initialValues = { datasetYaml: "" };
+type FormValues = typeof initialValues;
+
+// handle the common case where everything is nested under a `dataset` key
+interface NestedDataset {
+  dataset: Dataset[];
+}
+export function isDatasetArray(value: unknown): value is NestedDataset {
+  return (
+    typeof value === "object" &&
+    value != null &&
+    "dataset" in value &&
+    Array.isArray((value as any).dataset)
+  );
+}
+
+const DatasetYamlForm = () => {
+  const [createDataset] = useCreateDatasetMutation();
+  const toast = useToast();
+  const router = useRouter();
+
+  const handleCreate = async (
+    newValues: FormValues,
+    formikHelpers: FormikHelpers<FormValues>
+  ) => {
+    const { setErrors } = formikHelpers;
+    const parsedYaml = yaml.load(newValues.datasetYaml, { json: true });
+    let dataset: Dataset;
+    if (isDatasetArray(parsedYaml)) {
+      [dataset] = parsedYaml.dataset;
+    } else {
+      // cast to a Dataset and let the backend do validation
```

Comment:
I guess these casts are necessary because the [endpoint](https://github.com/ethyca/fides/blob/3680415b84e93053e8dbc78ce4488ab1f98dd3ff/clients/admin-ui/src/features/dataset/dataset.slice.ts#L52) only accepts that as an input type. I think if we're relying on the API to accept unknown values gracefully, that function should have `unknown` as its input type, which will make the casts unnecessary.